llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -1187,6 +1187,13 @@ void InstCombinerImpl::freelyInvertAllUsersOf(Value *I, Value *IgnoredUser) {
                        "canFreelyInvertAllUsersOf() ?");
     }
   }
+
+  // Adapt all dbg users
+  if (auto *Inst = dyn_cast<Instruction>(I); Inst && Inst->isUsedByMetadata()) {
+    auto *Not = Builder.CreateNot(I);
```

Comment:
I don't think we can have debug info create extra instructions, that can lead to codegen variations if debug info is enabled.

I think this needs to be encoded in a DWARF expression, not an IR instruction.

@jmorse is more expert at DWARF location information, though

Reply:
That's right, the presence of debug info may never affect the generated code. You probably want to create a DIExpression that has a DW_OP_not (and potentially modify the Verifier to allow this if this is the first time it's being introduced).